src/slider/styles.scss

```diff
@@ -0,0 +1,260 @@
+      2fr
+    )
+    3fr;
+  grid-auto-rows: 100%;
+  padding-block-start: awsui.$space-m;
+
+  // this keeps the labels to a maximum of two lines
+  > .label {
+    text-overflow: ellipsis;
+    display: -webkit-box;
+    -webkit-line-clamp: 2;
+    -webkit-box-orient: vertical;
+    overflow-y: clip;
+  }
+
+  &-noref {
+    grid-template-columns: 1fr 1fr;
+  }
+
+  &-reference {
+    grid-column-start: var(#{custom-props.$sliderReferenceColumn});
+    grid-column-end: var(#{custom-props.$sliderNextReferenceColumn});
+    grid-row: 1;
+    justify-self: center;
+    text-align: center;
+  }
+
+  &-min {
+    grid-column: 1;
+    grid-row: 1;
+    grid-column-end: var(#{custom-props.$sliderMinEnd});
+  }
+
+  &-max {
+    text-align: end;
+    justify-content: flex-end;
+    grid-row: 1;
+    grid-column-end: calc(var(#{custom-props.$sliderLabelCount}) + 1);
+    grid-column-start: var(#{custom-props.$sliderMaxStart});
+  }
+
+  @include styles.media-breakpoint-down(styles.$breakpoint-xx-small) {
+    & {
+      grid-template-columns: 1fr 1fr;
+    }
+
+    &-min {
+      grid-column: 1;
+    }
+
+    &-max {
+      grid-column: 2;
+    }
+
+    > .labels-reference {
+      display: none;
+    }
+  }
+}
+
+.ticks {
+  display: grid;
+  grid-template-columns: repeat(var(#{custom-props.$sliderLabelCount}), 1fr);
+  inline-size: 100%;
+  position: relative;
+  block-size: 0;
+}
+
+.tick {
+  grid-row: 1;
+  block-size: 8px;
+  inline-size: awsui.$space-xxxs;
+  padding-inline: 0;
+  padding-block: 0;
+  background: awsui.$color-background-slider-track;
+  inset-block-start: -4px;
+
+  &.filled {
+    background: awsui.$color-background-slider-handle-default;
+  }
+
+  &.error {
+    background: awsui.$color-text-status-error;
+  }
+
+  &.disabled {
+    background: awsui.$color-background-control-disabled;
+  }
+}
+
+.tooltip-thumb {
+  position: absolute;
+  inline-size: awsui.$space-xxl;
+  block-size: awsui.$space-xxl;
+  margin-block-start: awsui.$space-xs;
+}
+
+.thumb {
+  background: transparent;
+  appearance: none;
+  position: absolute;
+  block-size: 0;
+  inline-size: calc(100% + mixins.$thumb-size);
+  outline: 0;
+  margin-inline-start: calc(mixins.$thumb-size / -2);
+  margin-block-start: awsui.$space-xxs;
+  cursor: pointer;
+
+  &.disabled {
+    cursor: default;
+  }
+
+  // 1px added to min and max to cover the step line when it exists.
+  &.min {
+    margin-inline-start: -1px;
+  }
+
+  &.max {
+    margin-inline-start: -(mixins.$thumb-size) + 1;
+  }
+
+  &::-webkit-slider-thumb {
+    appearance: none;
+    @include mixins.base-thumb-styles();
+  }
+
+  &::-moz-range-thumb {
+    appearance: none;
+    @include mixins.base-thumb-styles();
+  }
+
+  &:hover::-webkit-slider-thumb {
+    @include mixins.base-thumb-hover-styles();
+  }
+
+  &:hover::-moz-range-thumb {
+    @include mixins.base-thumb-hover-styles();
+  }
+
+  &:focus::-webkit-slider-thumb,
```

Comment:
Looks like we're not using the usual `focus-visible` helpers. Is this intentional?

Reply:
The webkit-slider-thumb and moz-thumb elements are extremely finicky when it comes to pseudo elements like focus and hover. You have to define them separately, for example, or they won't work. I tried using the `focus-visible` helpers here but it broke the focus state. Is there any risk to not using it?

Reply:
The main purpose of the utility is being a kind of polyfill for `:focus-visible`, so it makes sure that we consistently show focus outlines when we detect a keyboard user. Right now you're kinda taking care of that using mouse and keyboard events.

The risk here would be that the behavior ends up being inconsistent in some edge cases.

Reply:
Slightly off-topic because it's design feedback but: I noticed that we're using different color shades to indicate focused state. Is there a reason why we're not using the standard focus outline color around the knob?

Reply:
This came directly from my sessions with visual design. We determined together that we liked the look of this better. Feedback noted though, we can keep an eye on it going forward.